### ISSUE TITLE
AWS S3 provider changes path behavior since recent updates.

### ISSUE THREAD

**Issue by blackskad** [travis-ci/travis-ci#9370]
Hi,

We noticed today that our build artifacts were no longer in the correct directory inside the s3 bucket. 

Expected location:
> s3://ourbucket/service-name/$TRAVIS_BRANCH/binary

Actual location:
> s3://ourbucket/service-name/$TRAVIS_BRANCH//home/travis/gopath/src/acquia/.../build/binary

I'm not sure why, but our travis file contained the $TRAVIS_BUILD_DIR variable as part of the local_dir option:

>    local_dir: "$TRAVIS_BUILD_DIR/build"

The docs never use that variable for the local_dir option, so I'm not sure why we did, but it worked like we expected until today. To achieve the same (and expected) behavior as before, it was enough to remove the $TRAVIS_BUILD_DIR from the local_dir option. I would expect both options to result in the same s3 location though, given that they're pointing to the same directory.

```
-    local_dir: "$TRAVIS_BUILD_DIR/build"
+    local_dir: "build"
```

While the issue itself is fixed for us with this change to our travis file, I wanted to make sure you are aware of this change in behavior.

**Comment by BanzaiMan** [comment]
When reporting a problem, please incl...

**Comment by BanzaiMan** [comment]
This issue should be fixed by `dpl` 1.9.4. See https://github.com/travis-ci/dpl/issues/786.

**Comment by blackskad** [comment]
Thank you for your reply. So it's not me removing $TRAVIS_BUILD_DIR from local_dir that fixed it, but the travis team reverting back to the old behavior. I should have run a new build before trying to fix it :) Thanks!
